src/com/google/javascript/jscomp/WhitespaceWrapGoogModules.java

```diff
@@ -0,0 +1,73 @@
+package com.google.javascript.jscomp;
+
+import com.google.common.base.Preconditions;
+import com.google.javascript.rhino.IR;
+import com.google.javascript.rhino.Node;
+
+public class WhitespaceWrapGoogModules implements HotSwapCompilerPass {
+
+  private final AbstractCompiler compiler;
+
+  WhitespaceWrapGoogModules(AbstractCompiler compiler) {
+    this.compiler = compiler;
+  }
+
+  @Override
+  public void process(Node externs, Node root) {
+    for (Node c = root.getFirstChild(); c != null; c = c.getNext()) {
+      Preconditions.checkState(c.isScript());
+      hotSwapScript(c, null);
+    }
+  }
+
+  @Override
+  public void hotSwapScript(Node scriptRoot, Node originalRoot) {
+    if (!isModuleFile(scriptRoot)) return;
+
+    // As per ClosureBundler:
+    /*
+	      // add the prefix on the first line so the line numbers aren't affected.
+	      out.append(
+	          "goog.loadModule(function(exports) {"
+	          + "'use strict';");
+	      append(out, Mode.NORMAL, contents);
+	      out.append(
+	          "\n" // terminate any trailing single line comment.
+	          + ";" // terminate any trailing expression.
+	          + "return exports;});\n");
+	      appendSourceUrl(out, Mode.NORMAL);
+     */
+
+    Node block = IR.block();
+    block.addChildToBack(IR.exprResult(IR.string("use strict"))); // needs to be explicit, to match ClosureBundler
+    for (Node c = scriptRoot.getFirstChild(); c != null; c = c.getNext()) {
+      block.addChildToBack(c.cloneTree());
```

Comment:
I don't think we need to clone the entire trees here. Just detaching them should be enough.